network_utils/src/network.rs

```diff
@@ -85,6 +85,14 @@ impl NetworkClient {
         }
     }
 
+    pub fn get_base_address(&self) -> &str {
```

Comment:
nit: I think the name style prefered is to just call the function the name of the member returned (ie. `base_address` or `base_port`) without the `get_`